### ISSUE TITLE
[webgl2] A separate package

### ISSUE THREAD

**Issue by stwind** [thi-ng/umbrella#337]
It is increasingly hard to use a single uniform compatible declaration for both WebGL 1 and 2 features (extensions, textures, UBO etc.). While WebGL2 [achieving pervasive support](https://www.khronos.org/blog/webgl-2-achieves-pervasive-support-from-all-major-web-browsers) now, would it be nice to have standalone package for WebGL2 only?

**Comment by postspectacular** [comment]
Yeah, the moment has finally come for WebGL2. Instead of separating v2 I'd propose to extract the legacy v1 stuff into own package... What are your thoughts?

**Comment by stwind** [comment]
Yeah extracting legacy stuffs to its own package seems to be a better idea. 

I am not quite sure how to do that yet, since there will be quite some overlaps/similarities between them. And as the "new" `webgl` package keep evolving to be more `WebGL2`, some common parts will almost eventually break for the legacy `webgl1` package. 

So the problem is how much or what should be kept in a common package, if any. Obviously the gl constants is the safest one to keep as common.
